### PR TITLE
Cw 284 softlayer api changes

### PR DESCRIPTION
**Pull request by SefiGiga**

### CODE REVIEW

File: cloudify-widget-common/pom.xml [hunk 1]
```diff
@@ -22,6 +22,37 @@
...
+            <version>4.11</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.mashape.unirest</groupId>
+            <artifactId>unirest-java</artifactId>
+            <version>1.3.27</version>
+        </dependency>
+        <dependency>
             <groupId>cloudify.widget</groupId>
```
Comment (guy-mograbi-at-gigaspaces) [inline cloudify-widget-common/pom.xml:56]: what about removing dependency on jclouds for softlayer?

File: cloudify-widget-pool-manager/src/main/resources/pool-manager-app-context.xml [hunk 2]
```diff
@@ -137,7 +137,25 @@
...
+    <bean id="pingAction" class="cloudify.widget.pool.manager.actions.PingAction"/>
+
+    <bean id="cloudServerApiFactory" class="cloudify.widget.pool.manager.CloudServerApiFactory"/>
+
+    <bean id="softlayerRestApi" class="cloudify.widget.softlayer.SoftlayerRestApi"/>
+
+
+    <bean id="softlayerCatalogManager" class="cloudify.widget.softlayer.SoftlayerCatalogManager">
+        <property name="stalePeriod" value="3600000"/>
+        <property name="pricesJSONArrayTemplate" value="[{'id':34807},{'id':27023},{'id':32500},{'id':32627},{'id':23070},{'id':35310},{'id':33483}]"/>
```
Comment (guy-mograbi-at-gigaspaces) [inline cloudify-widget-pool-manager/src/main/resources/pool-manager-app-context.xml:149]: What are these ids?

File: cloudify-widget-softlayer/src/main/java/cloudify/widget/softlayer/SoftlayerCatalogManager.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+     * and also include default price IDs required for the request to be successful.
+     *
+     * @param hardwareIds The comma delimited hardware IDs. For example "111,222,333,444".
+     * @param connectDetails
+     * @return a JSONArray of JSONObjects.
+     */
+    public JSONArray getPriceIds(String hardwareIds, SoftlayerConnectDetails connectDetails) {
+        Date now = new Date();
+
+        if (lastCatalogUpdate == null || now.getTime() - lastCatalogUpdate.getTime() >= getStalePeriod()) {
```
Comment (guy-mograbi-at-gigaspaces) [inline cloudify-widget-softlayer/src/main/java/cloudify/widget/softlayer/SoftlayerCatalogManager.java]: the logic if to update or not, should be in update function.

File: cloudify-widget-softlayer/src/main/java/cloudify/widget/softlayer/SoftlayerCloudServerApi.java [hunk 4]
```diff
@@ -115,97 +121,34 @@ public void setConnectDetails(SoftlayerConnectDetails connectDetails) {
...
-        String cloudProvider = CloudProvider.SOFTLAYER.label;
-        logger.info("building new context for provider [{}]", cloudProvider);
-        contextBuilder = ContextBuilder.newBuilder(cloudProvider)
-                .credentials(connectDetails.getUsername(), connectDetails.getKey())
-                .overrides(overrides)
-                .modules(modules);
-        context = contextBuilder.buildView(ComputeServiceContext.class);
-
-        return context;
+        // We're using REST, nothing to connect to.
```
Comment (guy-mograbi-at-gigaspaces) [inline cloudify-widget-softlayer/src/main/java/cloudify/widget/softlayer/SoftlayerCloudServerApi.java:127]: yey!!! nice comment.

File: cloudify-widget-softlayer/src/main/java/cloudify/widget/softlayer/SoftlayerRestApi.java [hunk 5]
```diff
@@ -0,0 +1,141 @@
+package cloudify.widget.softlayer;
```
Comment (guy-mograbi-at-gigaspaces) [inline cloudify-widget-softlayer/src/main/java/cloudify/widget/softlayer/SoftlayerRestApi.java:1]: please add documentation

File: cloudify-widget-pool-manager/src/main/resources/pool-manager-app-context.xml (hunk 2 above)
Comment (SefiGiga) [inline cloudify-widget-pool-manager/src/main/resources/pool-manager-app-context.xml:149]: Those are the price IDs that jclouds added for us without us knowing, besides the hardwareIDs specified in the pool

File: cloudify-widget-common/pom.xml (hunk 1 above)
Comment (SefiGiga) [inline cloudify-widget-common/pom.xml:56]: Currently it is still being used to run the bootstrap script.
I'll recheck.

**Pull request merged**